.github/workflows/cloudflare-pages.yml

```diff
@@ -0,0 +1,95 @@
+on:
+  # A push occurs to one of the matched branches.
+  push:
+    branches: [master]
+  # Or when a pull request event occurs for a pull request against one of the
+  # matched branches.
+  pull_request:
+    branches: [master]
+
+jobs:
+  cloudflare-pages:
+    # This name appears in GitHub's Checks API.
+    name: cloudflare-pages
+    # Do not trigger job for dependency update bot.
+    if: github.actor != 'renovate[bot]'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      # In progress comment will be updated with a deploy status.
+      - name: Add in progress comment to PR
+        uses: mshick/add-pr-comment@v2
+        with:
+          message: |
+            ## Deploying with <a href="https://pages.dev">Cloudflare Pages</a>
+            <table>
+              <tr>
+                <td><strong>Latest commit:</strong></td>
+                <td><code>${{ github.event.pull_request.head.sha }}</code></td>
+              </tr>
+              <tr>
+                <td><strong>Status:</strong></td><td>⏳ In progress</td>
+              </tr>
+            </table>
+      - name: Checkout code
+        uses: actions/checkout@v3
+        # Checkout pull request HEAD commit instead of merge commit.
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up Node.js 18
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: yarn
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Build app preview
+        run: yarn build-preview
+      - name: Publish to Cloudflare Pages
+        # Id is needed to access output in a next step.
+        id: cloudflare
+        uses: cloudflare/pages-action@v1
+        with:
+          apiToken: ${{ secrets.CLOUDFLARE_PAGES_API_TOKEN }}
+          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
+          # Project name created via Cloudflare dashboard.
+          projectName: oasis-explorer
+          # Use output directory of build-preview command.
+          directory: build
+      # On a subsequent run the original comment will be updated.
+      - name: Update PR comment
+        uses: mshick/add-pr-comment@v2
+        if: always()
+        with:
+          message: |
+            ## Deployed to <a href="https://pages.dev">Cloudflare Pages</a>
+            <table>
+              <tr>
+                <td><strong>Latest commit:</strong></td>
+                <td><code>${{ github.event.pull_request.head.sha }}</code></td>
+              </tr>
+              <tr>
+                <td><strong>Status:</strong></td><td>✅ Deploy successful!</td>
+              </tr>
+              <tr>
+                <td><strong>Preview URL:</strong></td>
+                <td><a href="${{ steps.cloudflare.outputs.url }}">${{ steps.cloudflare.outputs.url }}</a></td>
+               </tr>
+            </table>
+          message-failure: |
```

Comment:
We should have `message-cancelled` too
see: https://github.com/oasisprotocol/explorer/pull/289#issuecomment-1513297416
![image](https://user-images.githubusercontent.com/3758846/233371999-2cac8ac2-05fd-4222-8a4f-ae3a57f0bea0.png)

from a cancelled run: https://github.com/oasisprotocol/explorer/actions/runs/4752003232/jobs/8441821763